subsys/bluetooth/host/gatt.c

```diff
@@ -2852,12 +2855,18 @@ struct bt_gatt_attr *bt_gatt_find_by_uuid(const struct bt_gatt_attr *attr,
 					  const struct bt_uuid *uuid)
 {
 	struct bt_gatt_attr *found = NULL;
-	uint16_t start_handle = bt_gatt_attr_value_handle(attr);
-	uint16_t end_handle = start_handle && attr_count ?
-			      start_handle + attr_count : 0xffff;
+	uint16_t start_handle = bt_gatt_attr_get_handle(attr);
+	uint16_t end_handle = start_handle && attr_count
+				      ? MIN(start_handle + attr_count, BT_ATT_LAST_ATTRIBUTE_HANDLE)
+				      : BT_ATT_LAST_ATTRIBUTE_HANDLE;
+
+	if (attr != NULL && start_handle == 0U &&
+	    bt_uuid_cmp(attr->uuid, BT_UUID_GATT_PRIMARY) != 0) {
+		LOG_DBG("Could not find handle of attr %p", attr);
+		return NULL;
+	}
```

Comment:
I'm not sure what the point of this `if` is. Do we need it at all?

Reply:
I wanted to add additional verification that when attr != then start_handle should be different than 0, as a handle of 0 would indicate that the provided `attr` doesn't exist in the database, and thus cannot be used as a starting point for the search. 

However, the first attribute in the database, handle 0, shall also be possible to provide, and there's an assumption here (https://github.com/zephyrproject-rtos/zephyr/pull/83959#discussion_r1916785548) that this will always be a primary service

Reply:
> Do we need it at all?

Maybe not. It was meant to ensure that if the starting `attr` is not in the database, then it would return `NULL` when using it as a starting point. 

However, I see now that it only applies for static services, as `bt_gatt_attr_get_handle` will just return `attr->handle` for dynamically registered attributes, and thus it won't really work as intended. 

So we can either remove this check (more similar to previous iteration of the function), or we can verify that `attr` is part of the database before using it as a starting point for a search. This probably applies to several other GATT APIs that use the same "start from attr" approach.

Reply:
Update: The first attribute handle is 1, so if the handle returned by `bt_gatt_attr_get_handle` is 0, then the attribute is not in the database